### DISCUSSION TITLE
User implemented mapping method not taken into consideration

### DISCUSSION THREAD

**Discussion by AndreiLacatos** [riok/mapperly#1025] [Q&A]
I am using Mapperly to map objects used in gRPC contracts to domain objects. The protobuf entities are compiled into POCO object via protoc during application build.
When having an array of strings in the contract that has a list of Guid as the source, Mapperly does not seem to take into consideration the custom method inside the mapper class that maps a Guid to a string. In the generated mapper `Guid.ToString()` is called.

Using Mapperly 3.3.1-next.1 and .NET 8.

Proto:
```
message SampleResponse {
  string message = 1;
  repeated string items = 2;
}
```

Mapper:
```c#
[Mapper]
public static partial class GrpcMapper
{
    [MapperIgnoreSource(nameof(SampleModel.Id))]
    [MapProperty(nameof(SampleModel.ItemList), nameof(SampleResponse.Items))]
    public static partial SampleResponse MapGrpc(SampleModel source);

    private static string MapSpecial(Guid source) => $"hello/{source}";

    private static RepeatedField<string> MapSpecialArray(IEnumerable<Guid> source) => new()
    {
        source.Select(MapSpecial),
    };
}
```

Note: The presence of `MapSpecialArray` has no effect either

I've never had such problems until now, I suspect the problem is that the target type is source generated and user code.

Note 2: If I create a class which has a `RepeatedField<string>` property mappings work just fine.

Here is a repo to repro the issue: https://github.com/AndreiLacatos/GrpcSample

What am I missing here? How could I make it work?

**Comment by latonz** [comment]
Thank you for bringing this up, looks like a bug in Mapperly.
However, even with the bug fix it may not work as expected:
Repeated fields in proto don't have a setter, therefore you can only add / clear the collection but not set it. This is why Mapperly will never pick up `MapSpecialArray`. https://github.com/riok/mapperly/issues/783 may be what you need...
